File: frontend/app/vo/login/LoginVo.ts
/**
 * 인증 도메인 VO — backend 의 LoginVo / LoginAuthenticateRqVo / RsVo 와 1:1 매칭.
 *
 * 토큰 자체는 HttpOnly 쿠키로만 전달. 본 VO 의 응답은 식별 클레임만 노출.
 */

export interface LoginAuthenticateRqVo {
  loginId: string;
  password: string;
}

export interface LoginAuthenticateRsVo {
  accountSn: number;
  loginId: string;
  displayName: string;
  role: string;
}

export interface LoginSignupRqVo {
  loginId: string;
  password: string;
}

export interface LoginSignupRsVo {
  accountSn: number;
  loginId: string;
  displayName: string;
  role: string;
  createdAt: string;
}

export interface AuthMeRsVo {
  accountSn: number;
  loginId: string;
  displayName: string;
  role: string;
  createdAt: string;
  lastLoginDt: string | null;
}

File: backend/src/main/java/com/jsh/aidesk/serverapi/general/login/vo/LoginVo.java
package com.jsh.aidesk.serverapi.general.login.vo;

import java.time.OffsetDateTime;

import lombok.Data;

/**
 * 사용자 계정 VO (t_user) — 인증 + 조회 + insert 공통 캐리어.
 *
 * 컬럼 매핑은 LoginMapper.xml 참조. 다른 도메인이 user 식별만 쓸 거면 AuthenticatedUser 를 본다.
 */
@Data
public class LoginVo {

    private Long accountSn;
    private String loginId;
    private String password;
    private String displayName;
    private String role;
    private OffsetDateTime lastLoginDt;
    private OffsetDateTime createdAt;
    private OffsetDateTime updatedAt;
}
